### PR TITLE
Allow case insensitive inbox rules

### LINKED ISSUE #2309: Allow inbox rules to be case insensitive
* `/hey/` matches `hey` but not `HEY`
* `/HEY/` matches `HEY` but not `hey`
* `/hey/i` matches nothing, should match both

### PR DESCRIPTION
**Pull request by mlr46**
close #2309.

cc @vitoreiji

### CODE REVIEW

**Comment by mlr46** [comment]
I'm not sure why GitHub is not running the CI workflow. Do it, GitHub!

**Review by vitoreiji — CHANGES_REQUESTED** [review]
Thanks for this!

File: src/mail/InboxRuleHandler.js [hunk 1]
```diff
@@ -125,7 +125,15 @@ function _checkContainsRule(value: string, inboxRule: InboxRule): ?InboxRule {
 /** export for test. */
 export function _matchesRegularExpression(value: string, inboxRule: InboxRule): boolean {
 	if (isRegularExpression(inboxRule.value)) {
-		let regExp = new RegExp(inboxRule.value.substring(1, inboxRule.value.length - 1));
+		let flags = inboxRule.replace(/.*\/([gimsuy]*)$/, '$1');
+		let pattern = inboxRule.replace(new RegExp('^/(.*?)/' + flags + '$'), '$1');
+
+		// By default, we will make the inbox rule case insensitive.
```
[the remark this answers is not in the document]
Comment (mlr46) [reply src/mail/InboxRuleHandler.js:131]: Sounds good - will update to remove the default behaviour then and just use the flags if they exist.

File: src/mail/InboxRuleHandler.js [hunk 2]
```diff
@@ -125,7 +125,9 @@ function _checkContainsRule(value: string, inboxRule: InboxRule): ?InboxRule {
 /** export for test. */
 export function _matchesRegularExpression(value: string, inboxRule: InboxRule): boolean {
 	if (isRegularExpression(inboxRule.value)) {
-		let regExp = new RegExp(inboxRule.value.substring(1, inboxRule.value.length - 1));
+		let flags = inboxRule.replace(/.*\/([gimsuy]*)$/, '$1');
+		let pattern = inboxRule.replace(new RegExp('^/(.*?)/' + flags + '$'), '$1');
```
Comment (vitoreiji) [inline src/mail/InboxRuleHandler.js:129]: Sorry, I totally missed this on the first review.

```suggestion
		let flags = inboxRule.value.replace(/.*\/([gimsuy]*)$/, '$1');
		let pattern = inboxRule.value.replace(new RegExp('^/(.*?)/' + flags + '$'), '$1');
```

**Review by vitoreiji — CHANGES_REQUESTED** [review]
You need to get the `value` from the `inboxRule`.

File: src/mail/InboxRuleHandler.js (hunk 2 above)
Comment (mlr46) [reply src/mail/InboxRuleHandler.js:129]: Ah, yes. Copy pasta from fiddle.

**Review by vitoreiji — APPROVED** [review]
Works great, thanks!

**Pull request merged**